crates/storage/backend/rocksdb.rs

```diff
@@ -0,0 +1,294 @@
+use std::collections::{HashMap, HashSet};
+use std::path::Path;
+use std::sync::Arc;
+
+/// RocksDB backend
+#[derive(Debug)]
+pub struct RocksDBBackend {
+    /// Optimistric transaction database
+    db: Arc<OptimisticTransactionDB<MultiThreaded>>,
+}
+
+impl StorageBackend for RocksDBBackend {
+    fn open(path: impl AsRef<Path>) -> Result<Self, StoreError>
+    where
+        Self: Sized,
+    {
+        // Rocksdb optimizations options
+        let mut opts = Options::default();
+        opts.create_if_missing(true);
+        opts.create_missing_column_families(true);
+
+        opts.set_max_open_files(-1);
+        opts.set_max_file_opening_threads(16);
+
+        opts.set_use_fsync(false); // fdatasync
+
+        opts.enable_statistics();
+        opts.set_stats_dump_period_sec(600);
+
+        // Values taken from https://github.com/facebook/rocksdb/wiki/Setup-Options-and-Basic-Tuning#other-general-options
+        // These are 'real' default values
+        opts.set_level_compaction_dynamic_level_bytes(true);
+        opts.set_max_background_jobs(6);
+        opts.set_bytes_per_sync(1048576);
+        opts.set_compaction_pri(rocksdb::CompactionPri::MinOverlappingRatio);
+
+        // Open all column families
+        let existing_cfs = OptimisticTransactionDB::<MultiThreaded>::list_cf(&opts, path.as_ref())
+            .unwrap_or_else(|_| vec!["default".to_string()]);
+
+        let mut all_cfs_to_open = HashSet::new();
+        all_cfs_to_open.extend(existing_cfs.iter().cloned());
+        all_cfs_to_open.extend(TABLES.iter().map(|table| table.to_string()));
+
+        let cf_descriptors = all_cfs_to_open
+            .iter()
+            .map(|cf| {
+                let mut cf_opts = Options::default();
+                let mut bb_opts = BlockBasedOptions::default();
+                bb_opts.set_block_size(16 * 1024);
+                bb_opts.set_cache_index_and_filter_blocks(true);
+                bb_opts.set_pin_l0_filter_and_index_blocks_in_cache(true);
+                // Newest in this [list](https://github.com/facebook/rocksdb/blob/v8.6.7/include/rocksdb/table.h#L493-L521)
+                // We can check main
+                bb_opts.set_format_version(6);
+                cf_opts.set_block_based_table_factory(&bb_opts);
+                ColumnFamilyDescriptor::new(cf, cf_opts)
+            })
+            .collect::<Vec<_>>();
+
+        let db = OptimisticTransactionDB::<MultiThreaded>::open_cf_descriptors(
+            &opts,
+            path.as_ref(),
+            cf_descriptors,
+        )
+        .map_err(|e| StoreError::Custom(format!("Failed to open RocksDB with all CFs: {}", e)))?;
+        Ok(Self { db: Arc::new(db) })
+    }
+
+    fn create_table(&self, _name: &'static str, _options: TableOptions) -> Result<(), StoreError> {
+        // Now we are creating the tables in the open() function
+        // Check if this function is still needed
+        Ok(())
+    }
```

Comment:
I vote for removing this method. We can just assume all backends create the tables in the `open` methods.